core/execution/future/auction.go

```diff
@@ -61,6 +63,23 @@ func (m *Market) checkAuction(ctx context.Context, now time.Time, idgen common.I
 	}
 
 	if endTS := m.as.ExpiresAt(); endTS == nil || !endTS.Before(now) {
+		if isOpening && checkExceeded && m.as.ExceededMaxOpening(now) {
+			// cancel the market, exceeded opening auction
+			m.tradableInstrument.Instrument.Product.UnsubscribeTradingTerminated(ctx)
```

Comment:
settlement is unsubscribed in the call that actually cancels the market (does some other things like clearing the LPs, too)

Reply:
Same as comment below: good point, removed messy code-duplication, replaced with a single call to `terminateTrading` (had to split that up into 2 functions because `checkAuction` already acquires a mutex lock)